src/RustyObjectStore.jl

```diff
@@ -81,52 +98,203 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
+    )
+        (
+            !isnothing(storage_account_key)
+            && !isnothing(storage_sas_token)
+        ) && error("Should provide either a storage_account_key or a storage_sas_token")
+
+        params = copy(opts.params)
+
+        params["azure_storage_account_name"] = storage_account_name
+        params["azure_container_name"] = container_name
+
+        if !isnothing(storage_account_key)
+            params["azure_storage_account_key"] = storage_account_key
+        end
+
+        if !isnothing(storage_sas_token)
+            params["azure_storage_sas_token"] = storage_sas_token
+        end
+
+        if !isnothing(host)
+            params["azurite_host"] = host
+        end
+
+        return new(Config("az://$(container_name)/", params))
+    end
+end
+
+as_config(wrapper::AzureConfig) = wrapper.config
+
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `region::String`: AWS S3 region.
+- `bucket_name::String`: AWS S3 bucket name.
+- `access_key_id::Option{String}`: (Optional) AWS S3 access key id.
+- `secret_access_key::Option{String}`: (Optional) AWS S3 secret access key.
+- `session_token::Option{String}`: (Optional) AWS S3 session_token.
+- `host::Option{String}`: (Optional) Alternative S3 host. For example, if using Minio.
+- `opts::ClientOptions`: (Optional) Client configuration options.
+"""
+struct AwsConfig <: AsConfig
+    config::Config
+    function AwsConfig(;
+        region::String,
+        bucket_name::String,
+        access_key_id::Option{String} = nothing,
+        secret_access_key::Option{String} = nothing,
+        session_token::Option{String} = nothing,
+        host::Option{String} = nothing,
+        opts::ClientOptions = ClientOptions()
+    )
+        params = copy(opts.params)
+
+        params["region"] = region
+        params["bucket_name"] = bucket_name
+
+        if !isnothing(access_key_id)
+            params["aws_access_key_id"] = access_key_id
+        end
+
+        if !isnothing(secret_access_key)
+            params["aws_secret_access_key"] = secret_access_key
+        end
+
+        if !isnothing(session_token)
+            params["aws_session_token"] = session_token
+        end
+
+        if !isnothing(host)
+            params["minio_host"] = host
+        end
+
+        return new(Config("s3://$(bucket_name)/", params))
+    end
+end
+
+as_config(wrapper::AwsConfig) = wrapper.config
+
+function Base.show(io::IO, config::Config)
+    dict = copy(JSON3.read(config.config_string))
+    print(io, "Config(")
+    first = true
+    for key in keys(dict)
+        if first
+            first = false
+        else
+            print(io, ", ")
+        end
+        if occursin(r"secret|token|key", string(key))
+            print(io, repr(key), " => ", repr("*****"))
+        else
+            print(io, repr(key), " => ", repr(dict[key]))
+        end
+    end
     print(io, ")")
 end
 
-const _AzureCredentialsFFI = NTuple{4,Cstring}
+const _ConfigFFI = NTuple{1,Cstring}
```

Comment:
Why do we need this to be a `Tuple{CString}` rather than just a `CString`?

Reply:
Will change!